tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -0,0 +1,354 @@
+set -o pipefail
+
+kubernetes_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+# shellcheck disable=1091
+source "${kubernetes_dir}/../../gha-run-k8s-common.sh"
+
+# Where the kbs sources will be cloned
+readonly COCO_KBS_DIR="/tmp/kbs"
+# The k8s namespace where the kbs service is deployed
+readonly KBS_NS="coco-tenant"
+# The kbs service name
+readonly KBS_SVC_NAME="kbs"
+
+# Set "allow all" policy.
+#
+kbs_set_allow_all_policy() {
+	kbs_set_policy "$COCO_KBS_DIR/kbs/sample_policies/allow_all.rego"
+}
+
+# Set policy.
+#
+# Parameters:
+#	$1 - path to policy file
+#
+kbs_set_policy() {
+	local file="${1:-}"
+
+	if [ ! -f "$file" ]; then
+		echo "ERROR: policy file '$file' does not exit"
+		return 1
+	fi
+
+	kbs_install_cli
+
+	kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
+		--auth-private-key "$(kbs_get_private_key)" set-resource-policy \
+		--policy-file "$file"
+}
+
+# Set resource data.
+#
+# Parameters:
+#	$1 - repository name (optional)
+#	$2 - resource type (mandatory)
+#	$3 - tag (mandatory)
+#	$4 - resource data
+#
+# Note: kbs-client will be installed if not yet.
+#
+kbs_set_resource() {
+	local repository="${1:-}"
+	local type="${2:-}"
+	local tag="${3:-}"
+	local data="${4:-}"
+	local file
+	local rc=0
+
+	if [ -z "$data" ]; then
+		echo "ERROR: missing data parameter"
+		return 1
+	fi
+
+	file=$(mktemp -t kbs-resource-XXXXX)
+	echo "$data" > "$file"
+
+	kbs_set_resource_from_file "$repository" "$type" "$tag" "$file" || \
+		rc=$?
+
+	rm -f "$file"
+	return $rc
+}
+
+# Set resource, read data from file.
+#
+# Parameters:
+#	$1 - repository name (optional)
+#	$2 - resource type (mandatory)
+#	$3 - tag (mandatory)
+#	$4 - resource data
+#
+# Note: kbs-client will be installed if not yet.
+#
+kbs_set_resource_from_file() {
+	local repository="${1:-}"
+	local type="${2:-}"
+	local tag="${3:-}"
+	local file="${4:-}"
+
+	if [[ -z "$type" || -z "$tag" ]]; then
+		echo "ERROR: missing type='$type' and/or tag='$tag' parameters"
+		return 1
+	elif [ ! -f "$file" ]; then
+		echo "ERROR: resource file '$file' does not exist"
+		return 1
+	fi
+
+	local path=""
+	[ -n "$repository" ] && path+="${repository}/"
+	path+="${type}/"
+	path+="${tag}"
+
+	kbs_install_cli
+
+	kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
+		--auth-private-key "$(kbs_get_private_key)" set-resource \
+		--path "$path" --resource-file "$file"
+}
+
+# Return the private key file used for CLI authentication.
+#
+kbs_get_private_key() {
+	echo "$COCO_KBS_DIR/kbs/config/kubernetes/base/kbs.key"
+}
+
+# Build and install the kbs-client binary, unless it is already present.
+#
+#
+kbs_install_cli() {
+	command -v kbs-client >/dev/null && return
+
+	if ! command -v apt >/dev/null; then
+		echo "ERROR: running on unsupported distro"
+		return 1
+	fi
+
+	local pkgs="build-essential"
+	command -v rustc >/dev/null || pkgs+=" rustc"
+	command -v cargo >/dev/null || pkgs+=" cargo"
+
+	# shellcheck disable=2086
+	sudo apt-get install -y $pkgs
+
+	pushd "$COCO_KBS_DIR"
```

Comment:
I think you need to be in `$COCO_KBS_DIR`/KBS for the make below to work.

Reply:
Good catch! Fixed!